### PR TITLE
Using entity for `$this->Form->create()`

### PR DESCRIPTION
**Pull request by groovenectar**
Fixes: https://github.com/CakeDC/users/issues/854: "Reset password -- No context provider found for value of type string"

### CODE REVIEW

**Comment by groovenectar** [comment]
I see this is also present in `social_email.php`, but I'm not aware of how to test that myself yet as I haven't gotten to 3rd party logins yet

**Comment by rochamarcelo** [comment]
The other occurrences should be fixed.

**Comment by groovenectar** [comment]
@rochamarcelo I've added the change to `social_email.php` as well, but please note that I'm not able to test that one myself at this time

File: src/Loader/MiddlewareQueueLoader.php [hunk 1]
```diff
@@ -115,7 +115,11 @@ protected function loadAuthorizationMiddleware(MiddlewareQueue $middlewareQueue,
         }
 
         $middlewareQueue->add(new AuthorizationMiddleware($plugin, Configure::read('Auth.AuthorizationMiddleware')));
-        $middlewareQueue->add(new RequestAuthorizationMiddleware());
+        
+        if (Configure::read('Auth.AuthorizationMiddleware.requireAuthorizationCheck') !== false) {
```
Comment (rochamarcelo) [inline src/Loader/MiddlewareQueueLoader.php]: I don't like this key naming. The condition is not for AuthorizationMiddleware but RequestAuthorizationMiddleware

Comment (rochamarcelo) [reply src/Loader/MiddlewareQueueLoader.php]: @steinkel ?

Comment (rochamarcelo) [reply src/Loader/MiddlewareQueueLoader.php]: Additionally you could just replace this loader with Users.middlewareQueueLoader config

Comment (steinkel) [reply src/Loader/MiddlewareQueueLoader.php]: I agree, naming could be improved

**Comment by rochamarcelo** [comment]
Thank you @groovenectar

**Pull request merged**